changelog.d/gh-5633.fixed

```diff
@@ -0,0 +1,2 @@
+Semgrep threw list index out of range error in the text formatter when the dedented_lines list was empty.
+This PR adds a conditions to factor for that and fixes the relevant issue.
```

Comment:
Two issues with this changelog entry:

- our users don't read semgrep's source code, so they wouldn't know what "dedented_lines list" means
- when this line is published as part of the changelog, it's unclear what "This PR" is referring to

```suggestion
Semgrep used to crash when trying to print findings that match only whitespace, such as when a rule disallows two newlines at the end of a file. This crash is now fixed.
```